CONTRIBUTING.md

```diff
@@ -38,6 +38,70 @@ with `gn format`.
 [objcc_style]: https://google.github.io/styleguide/objcguide.html#objective-c
 [java_style]: https://google.github.io/styleguide/javaguide.html
 
+## Testing
+
+The testing policy for contributing to the flutter engine can be found at the [Tree Hygiene Wiki][tree_hygiene_wiki]. The summary is that all PR's to the engine should be tested or have an explicit test exemption.
+
+Because the engine targets multiple platforms the testing infrastructure is fairly complicated. Here are some more resources to help guide writing tests:
+
+- [Testing the engine wiki][testing_the_engine_wiki] - A guide on writing tests for the engine including an overview of the different tests and the different technologies the engine uses.
+- [//testing](./testing) - This is where the `run_tests.py` script is located. All tests will have the ability to be executed with `run_tests.py`.
+- [//ci/builders](./ci/builders) - The JSON files that determine how tests are executed on CI.
+
+Tests will be executed on CI, but some tests will be executed before PR's can be merged (presubmit) and others after they have been merged (postsubmit). Ideally everything would be presubmit but tests that take up more resources are executed in postsubmit.
+
+### Example run_tests.py invocation
+
+```sh
+# Configure host build for macOS arm64 debug.
+$ flutter/tools/gn --runtime-mode=debug --unoptimized --no-lto --mac-cpu=arm64
+# Compile default targets (should cover all applicable run_tests.py requirements).
+$ ninja -j100 -C out/host_debug_unopt_arm64
+# Run all cross-platform C++ tests for the debug build arm64 variant.
+$ cd flutter/testing
+$ ./run_tests.py --variant=host_debug_unopt_arm64 --type=engine`
+```
+
+### Directory
```

Comment:
This seems valuable.  I'm not 100% sold this is the right place.  Maybe into the wiki?  I ran out of gas updating it.  I want to make sure it's something we want.

Reply:
I think this is a better place than the wiki since it will be versioned with changes in the engine. But either way is fine TBH. The wiki has a lower barrier to editing.

Reply:
Random thought for later: Maybe we could we just put these descriptions directly in `run_tests.py` and spit them out with a `--list` argument?

Reply:
The code could even be refactored to be data driven so we had the list in code.

Reply:
You don't even need the list in code. Let GN tell you what the test executables are. For instance `gn desc out/host_debug_unopt_arm64/ :default deps --all --type=executable --testonly=true --as=output` will tell their names.

Add `--format=json` to make it machine readable.

We could also create GN templates to add more metadata about the test that `gn desc` can dump out. IIRC, that is how the new build rules work as well.